nixos/modules/services/web-servers/quakejs.nix

```diff
@@ -0,0 +1,207 @@
+        type = port;
+        default = 27960;
+        description = ''
+          Port for the server to listen on.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = lines;
+        description = ''
+          Extra configuration options. Note that options changed via RCON will not be persisted. To list all possible options, use “cvarlist 1” via RCON.
+        '';
+        default = ''
+          set dedicated 2
+          seta sv_hostname "QuakeJS NixOS Server"
+          seta timelimit 30
+          seta fraglimit 30
+          seta g_inactivity 3000
+          seta g_forcerespawn 0
+          set d1 "map q3dm7 ; set nextmap vstr d2"
+          set d2 "map q3dm17 ; set nextmap vstr d1"
+          vstr "d1"
+        '';
+        example = literalExpression ''
+          seta g_motd "Happy fragging!";
+          seta g_gametype 0
+          set fs_game "baseq3"
+          set fs_cdn "content.quakejs.com"
+        '';
+      };
+
+    };
+
+  };
+
+  config = {
+
+    systemd.services.quakejs-ds = mkIf (cfg.dedicated-server.enable) {
+      description = "Dedicated server for QuakeJS";
+      environment = mkIf (cfg.eula) { EULAACCEPTED = "true"; };
+      serviceConfig = {
+        ExecStart = "${pkgs.quakejs}/bin/quakejs-ded +set net_port ${toString cfg.dedicated-server.port} +exec ${pkgs.writeText "server.cfg" cfg.dedicated-server.extraConfig}";
+        WorkingDirectory = "/var/lib/quakejs";
+        StateDirectory = [ "quakejs" ];
+        Type = "simple";
+        DynamicUser = true;
+      };
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+    };
+
+    systemd.services.quakejs = mkIf (cfg.enable) {
+      description = "Game client for QuakeJS";
+      serviceConfig = {
+        ExecStart = "${pkgs.quakejs}/bin/quakejs  --config ${format.generate "web.json" cfg.settings}";
+        WorkingDirectory = "/var/lib/quakejs";
+        StateDirectory = [ "quakejs" ];
+        Type = "simple";
+        DynamicUser = true;
+      };
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+    };
+
+    systemd.services.quakejs-cs = mkIf (cfg.content-server.enable) {
+      description = "Content server for QuakeJS";
+      serviceConfig = {
+        ExecStart = "${pkgs.quakejs}/bin/quakejs-content  --config ${format.generate "config.json" cfg.content-server.settings}";
+        WorkingDirectory = "/var/lib/quakejs";
+        StateDirectory = [ "quakejs" ];
+        Type = "simple";
+        DynamicUser = true;
+      };
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+    };
+
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts =
+        optionals (cfg.enable) [ cfg.settings.port ]
+        ++ optionals (cfg.content-server.enable) [ cfg.content-server.settings.port ]
+        ++ optionals (cfg.dedicated-server.enable) [ cfg.dedicated-server.port ];
+    };
+
+    services.caddy = {
+      enable = true;
```

Comment:
The default web server we usually use in nixos is nginx. Also right now this module would uncondtionally enable caddy.